### DISCUSSION TITLE
log format option

### DISCUSSION THREAD

**Discussion by magnus919** [zwave-js/zwave-js#4416] [Ideas]
Any chance we might get a **log format** option, please? For those using log monitoring like Grafana Loki, Splunk, or ELK stack it would be very useful to ship json-formatted logs.

**Comment by AlCalzone** [comment]
See https://github.com/zwave-js/zwavejs2mqtt/issues/2136#issuecomment-1004817926 - there is a json (or logfmt) transport available, which can be hooked into the driver. Documentation is a bit lacking though, I hope I'll get to it soon.

**Comment by magnus919** [comment]
I'm not really sure what to do with that. It look like one must be a Javascript developer to use any of that if I understand what I've been looking at for the past hour. I think I was looking for a feature where a mere mortal could save logs in JSON format.

**Reply by AlCalzone** [reply]
Which application are you using?

**Reply by magnus919** [reply]
zwave-js...

**Reply by AlCalzone** [reply]
So you're using this library directly? Or the Homeassistant addon?

**Reply by magnus919** [reply]
I'm running `zwave-js` in its own Docker container, in conjunction to an adjacent `homeassistant` container on the same host. I'm not using the Supervised version of `homeassistant`.

**Reply by AlCalzone** [reply]
I think I'm being dense here, but the names around the ecosystem are easy to be confused. I'm not sure what exactly you mean by `zwave-js`:
- This is `node-zwave-js` (available as the npm packet `zwave-js`), the **driver** library, which talks to the stick. This library is used by applications to provide Z-Wave functionality.
- Then there's [`zwavejs2mqtt`](https://github.com/zwave-js/zwavejs2mqtt), the semi-official management UI/dashboard
- [`zwave-js-server`](https://github.com/zwave-js/zwave-js-server) is a wrapper around `node-zwave-js`, which is how HomeAssistant talks to the driver
- And last but not least, there's https://www.home-assistant.io/integrations/zwave_js/, the Home Assistant integration, which can either be used with HomeAssistant's official Z-Wave JS addon, or the “Z-Wave JS to MQTT” addon which hosts the aforementioned `zwavejs2mqtt` dashboard.
